macos/FluentUI/Separator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+@objc(MSFSeparatorStyle)
+public enum SeparatorStyle: Int {
+    case `default`
+    case shadow
+
+    fileprivate var color: NSColor {
+		switch self {
+		case .default:
+			if #available(OSX 10.14, *) {
+				return NSColor.separatorColor
+
+			} else {
+				return NSColor.systemGray
+			}
+		case .shadow:
+			return NSColor.gridColor
+        }
+    }
+}
+
+@objc(MSFSeparatorOrientation)
+public enum SeparatorOrientation: Int {
+    case horizontal
+    case vertical
+}
+
+
+@objc(MSFSeparator)
+open class Separator: NSView {
+	private var orientation: SeparatorOrientation = .horizontal
+	
+	private var style: SeparatorStyle = .default
+
+	@objc public override init(frame frameRect: NSRect) {
+		super.init(frame: frameRect)
```

Comment:
why let it available?

Reply:
Can also remove the initWithFrame override altogether and force clients to specify the orientation. That makes a little more sense to me

Reply:
Also good to point out I need to create the markdown documentation file as part of this review :D

Reply:
And also just code documentation. At the least everything public needs documentation, ideally just everything.

Reply:
Code documentation added. Markdown still pending